utils/cq/d_logging_check.py

```diff
@@ -141,6 +141,17 @@ def __next__(self):
 PREFIXES_ALL = PREFIXES.copy()
 PREFIXES_ALL.extend(PREFIXES_NNL)
 
+# Variables that are checked for when %s is using in a logging macro.  The number of %s calls
+# is counted, and the number of times any of the strings below is included is also counted and
+# if the two are the same then no warning is issued.  This allows us to skip checks for cases where
+# strings are of known types, for example uris or CaRT group names, or performing enum->string
+# lookup.
+ALLOWED_VARS = ['strerror', 'db_file', 'ii_group_name', 'cg_grpid', 'agg_op2str', 'gc_name',
```

Comment:
so what is the purpose of checking variable names?  is this to avoid PII leak?

Reply:
> is this to avoid PII leak?

Yes.

> so what is the purpose of checking variable names?

Some variable names are known to contain things like cart group name or URIs so these don't need to be manually checked or annotated.  It's a (crude) attempt to reduce the number of lines we need to audit/modify/annotate.

This is really just doing string matching on the whole line so it's kinda dumb but it does work to catch a lot of cases. We also have quite a number of places and ways where either arrays or functions are used to convert enums to strings and there are covered here as well..